### ISSUE TITLE
YAML file with P6 head for large object in OBB model of YOLO26

### ISSUE THREAD

**Issue by LeMinhNgan** [ultralytics/ultralytics#23325]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello,

I need to train a large object that is about 80% of image frame. Seem the P5 head does not work. Could you please advise YAML file with P6 head of YOLO26 for large object? Thank

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeMinhNgan, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like model head / YAML architecture changes for OBB), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). In particular, sharing your current model YAML (or a snippet), image sizes, and a couple representative samples of the “large object ~80% of frame” cases will help the team advise properly 🧩👀.

This is an automated response, and an Ultralytics engineer will also assist soon.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
What's the `imgsz` you used?

**Comment by LeMinhNgan** [comment]
> What's the `imgsz` you used?

it's 2560, since i need pixle-wise precision

**Comment by LeMinhNgan** [comment]
Seem the OBB model still does not work well with large object. Can Anyone advise me how to fine tune  the YAML for large object in OBB of YOLO26?

**Comment by glenn-jocher** [comment]
YOLO26 OBB should generally handle objects that occupy most of the frame without needing a P6 head, so if it “doesn’t work well” it’s usually either label formatting/angle conventions or overly aggressive spatial augmentation; please double-check your labels follow the 4-corner YOLO OBB format and that angles are constrained to 0–90° as noted in the [OBB task docs](https://docs.ultralytics.com/tasks/obb/), then try training closer to the pretrained resolution (e.g. `imgsz=1024`) and do high-res inference at `imgsz=2560` while disabling mosaic-heavy aug that tends to crop/distort large objects, e.g.:  
```bash
yolo obb train model=yolo26l-obb.pt data=your.yaml imgsz=1024 mosaic=0.0 mixup=0.0 copy_paste=0.0 scale=0.0 translate=0.0
```  
If you can paste `yolo checks` output and share 2–3 representative images + their `*.txt` labels (or a tiny repro dataset), we can recommend more targeted config/YAML changes (see also the [model YAML configuration guide](https://docs.ultralytics.com/guides/model-yaml-config/)).

**Comment by LeMinhNgan** [comment]
> YOLO26 OBB should generally handle objects that occupy most of the frame without needing a P6 head, so if it “doesn’t work well” it’s usually either label formatting/angle conventions or overly aggressive spatial augmentation; please double-check your labels follow the 4-corner YOLO OBB format and that angles are constrained to 0–90° as noted in the [OBB task docs](https://docs.ultralytics.com/tasks/obb/), then try training closer to the pretrained resolution (e.g. `imgsz=1024`) and do high-res inference at `imgsz=2560` while disabling mosaic-heavy aug that tends to crop/distort large objects, e.g.:
> 
> yolo obb train model=yolo26l-obb.pt data=your.yaml imgsz=1024 mosaic=0.0 mixup=0.0 copy_paste=0.0 scale=0.0 translate=0.0
> If you can paste `yolo checks` output and share 2–3 representative images + their `*.txt` labels (or a tiny repro dataset), we can recommend more targeted config/YAML changes (see also the [model YAML configuration guide](https://docs.ultralytics.com/guides/model-yaml-config/)).

Thank for that. Please find the attachment for my data images and label

<img width="4282" height="3260" alt="Image" src="https://github.com/user-attachments/assets/a0923300-c074-4eee-8323-b9a7c6ddf098" />
[251027_c1_0000000003_rotated_m.txt](https://github.com/user-attachments/files/24756459/251027_c1_0000000003_rotated_m.txt)
<img width="4324" height="3322" alt="Image" src="https://github.com/user-attachments/assets/1f526b8a-f856-480b-aff7-5118d07e9d10" />
[251027_c1_0000000026_rotated_m.txt](https://github.com/user-attachments/files/24756460/251027_c1_0000000026_rotated_m.txt)

**Comment by glenn-jocher** [comment]
Thanks for sharing—GitHub attachments don’t render reliably on my side; can you paste the raw contents (first ~3 lines) of each `*.txt` label here as text? The most common cause of “large object” OBB failures is that the 4-corner coordinates are still in pixel units (or out of order) rather than normalized `[0,1]` in `class x1 y1 x2 y2 x3 y3 x4 y4` format as shown in the [YOLO OBB dataset format docs](https://docs.ultralytics.com/datasets/obb/).

If you want a quick sanity-check locally, run this on one of your label files and confirm `mn>=0` and `mx<=1`:

```python
import numpy as np, pathlib

p = pathlib.Path("251027_c1_0000000003_rotated_m.txt")
a = np.loadtxt(p, ndmin=2)
xy = a[:, 1:].reshape(-1)
print("mn:", float(xy.min()), "mx:", float(xy.max()), "shape:", a.shape)
```

**Comment by LeMinhNgan** [comment]
> Thanks for sharing—GitHub attachments don’t render reliably on my side; can you paste the raw contents (first ~3 lines) of each `*.txt` label here as text? The most common cause of “large object” OBB failures is that the 4-corner coordinates are still in pixel units (or out of order) rather than normalized `[0,1]` in `class x1 y1 x2 y2 x3 y3 x4 y4` format as shown in the [YOLO OBB dataset format docs](https://docs.ultralytics.com/datasets/obb/).
> 
> If you want a quick sanity-check locally, run this on one of your label files and confirm `mn>=0` and `mx<=1`:
> 
> import numpy as np, pathlib
> 
> p = pathlib.Path("251027_c1_0000000003_rotated_m.txt")
> a = np.loadtxt(p, ndmin=2)
> xy = a[:, 1:].reshape(-1)
> print("mn:", float(xy.min()), "mx:", float(xy.max()), "shape:", a.shape)

Hello,
image1:
0 0.1581977903213129 0.13146238509126695 0.8169287113379815 0.07020003347795926 0.8592557994706523 0.8554192229038854 0.20052487845398387 0.9166815745171932

image2:
0 0.2088433562300299 0.08609592983473045 0.8601651807526409 0.1498406111630025 0.8156799259944496 0.9199277543648404 0.16435810147183866 0.8561830730365684

**Comment by glenn-jocher** [comment]
Those label lines are already normalized to `[0,1]`, so the next most common OBB failure is **point ordering** (the 4 corners must be in a consistent clockwise order and not “bow-tied”/self-intersecting); can you run this quick overlay check and confirm the drawn quad matches the object exactly (see the expected format in the [YOLO OBB dataset format docs](https://docs.ultralytics.com/datasets/obb/))?

```python
import cv2, numpy as np

im_path = "251027_c1_0000000003_rotated_m.jpg"  # update
lb_path = "251027_c1_0000000003_rotated_m.txt"  # update

im = cv2.imread(im_path)
h, w = im.shape[:2]

a = np.loadtxt(lb_path, ndmin=2)
pts = a[:, 1:].reshape(-1, 4, 2)
pts[..., 0] *= w
pts[..., 1] *= h
pts = pts.astype(np.int32)

for p in pts:
    cv2.polylines(im, [p], isClosed=True, color=(0, 255, 0), thickness=6)

cv2.imwrite("obb_check.jpg", im)
print("Wrote obb_check.jpg")
```

If `obb_check.jpg` looks correct, please paste `yolo checks` plus your exact `yolo obb train ...` command (including all aug args) and a screenshot of `runs/obb/train/results.png`, and we can narrow down whether this is augmentation/resize behavior vs. something that warrants an architecture/YAML change.
